app/routes.py

```diff
@@ -1,2 +1,338 @@
+        else:
+            form_data = request.get_json()
+
+            goal.title = form_data["title"]
+
+            db.session.commit()
+
+        # return make_response(f"Book #{book.id} successfully updated")
+            return {
+                    "goal": {
+                        "id": goal.goal_id,
+                        "title": goal.title
+                    }
+                }
+   
+    if request.method == "DELETE":
+        if goal is None:
+            return jsonify(None), 404
+        # request_body = request.get_json() request body contains information to help conduct the actual request
+        # ie post request requires data to determines how to actually create the task
+        db.session.delete(goal)
+        db.session.commit()
+
+        return {"details": f'Goal {goal.goal_id} "{goal.title}" successfully deleted'}
+
+@goals_bp.route("/<goal_id>/tasks", methods= ["GET", "POST"])
+def handle_goal_tasks(goal_id):
+#we always pass goal id because every task is related to a goal
+#the goal is the one, tasks are the many
+
+    goal = Goal.query.get(goal_id)
+    if goal is None:
+        # return "Author not found", 404
+        return jsonify(None), 404
+
+    if request.method == "GET":
+        tasks_response = []
+        for task in goal.tasks:
+            tasks_response.append(
+                {
+                "id": task.task_id, #references variable from line 286
+                "goal_id": task.goal_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": False if task.completed_at == None else True
+                }
+            )
+        goal_response = {
+            "id": goal.goal_id,
+            "title": goal.title,
+            "tasks": tasks_response
+        }
+    
+        return jsonify(goal_response)
+        
+    elif request.method == "POST":
+        request_body = request.get_json()#takes request_body and transforms out of json format into python dictionaries and lists
+        # requested_goal = Goal.query.get(id=goal_id)
+        task_ids= []
+        for task_id in request_body["task_ids"]:
+            task = Task.query.get(task_id)
+            task.goal_id = goal_id
+            task_ids.append(task.task_id)
+        
+        db.session.commit()
+
+        return {
+            "id": goal.goal_id,
+            "task_ids": task_ids
+        }
+            # new_task = Task(title=request_body["title"],
+            #                 description=request_body["description"],
+            #                 completed_at=request_body["completed_at"])
+
+        db.session.add(new_task)
+        #used when creating new record
+        db.session.commit()
+        #used when we change any of the records, we want to commit before returning, commit as soon as you expect the changes to persist in the database
+
+        return make_response(f"Task {new_task.title} by {new_task.goal.name} successfully created", 201)
```

Comment:
`make_response` returns data in HTML form and would require the client additional syntax to access this text directly or turn into JSN. We can instead use `jsonify` return JSON data instead.

```suggestion
        return jsonify(f"Task {new_task.title} by {new_task.goal.name} successfully created"), 201
```